### PR TITLE
ARTEMIS-1462 Allow ActiveMQScheduledComponent initial delay configuration

### PR DESCRIPTION
**Pull request by franz1981**
It contains:
 - an improved documentation of the constructors
 - the initial delay configuration

### CODE REVIEW

File: artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java [hunk 1]
```diff
@@ -165,4 +165,37 @@ public void run() {
...
+         @Override
+         public void run() {
+            latch.countDown();
+         }
+      };
+      final long start = System.nanoTime();
+      local.start();
+      try {
+         final boolean triggeredBeforePeriod = latch.await(local.getPeriod(), local.getTimeUnit());
+         final long timeToFirstTrigger = TimeUnit.NANOSECONDS.convert(System.nanoTime() - start, local.getTimeUnit());
```
Comment (clebertsuconic) [inline artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java]: do you actually use NANOSECONDS on the Lock?

Comment (franz1981) [reply artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java]: Depends on the case but there are parts where I've used it: It is immune to ntpd/clock changes...

Comment (clebertsuconic) [reply artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java]: where you need NANOSECONDS? scheduling at nanosecond may kill perf.. where is it?

Comment (franz1981) [reply artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java]: I'm not scheduling at nanos but measuring elapsed time in nanoseconds, it is different :)
The scheduling is in seconds or milliseconds in some tests.

**Pull request closed without merging**